### DISCUSSION TITLE
[Risolto] Integrare Gateway bticino in Home Assistant con Homekit

### DISCUSSION THREAD

**Discussion by CiccioDroid** [indomus/forum#650] [HUB: Home Assistant]
Ciao a tutti,
Mi sono informato molto prima di scrivere, ma non ho trovato risposte recenti a questa domanda. 
Vorrei integrare i componenti bticino living now in HA usando l'integrazione di homekit, questo perché voglio mantenere il cloud bticino e al contempo vedere tutto in Home Assistant. Ho letto, però, che con homekit non riuscirò a vedere i consumi delle prese connesse e questo è un problema visto che ho già 4 prese connesse zigbee di altro produttore. 
Il problema, invece, non si dovrebbe presentare se collego le prese direttamente al coordinatore zigbee, ma perderei le funzioni cloud di bticino e anche 80 euro di Gateway. 
Avete consigli?
Grazie

**Comment by snakuzzo** [comment]
Ne abbiamo discusso proprio questa mattina sulla chat Telegram.
Ti riporto quanto scritto da @sdomotica

_1) Sganciando il gateway ed usando le pennette (pro-gratis contro-non ha l'applicazione ufficiale/cloud etc, non hai i comandi scenari)
2) Integrazione Homekit (pro-gratis e va senza internet e hai ancora l'app ufficiale, contro-perdi l'integrazione nativa Homekit e non hai i consumi ed i comandi scenari
3) Integrazione nativa HA Legrand (pro-gratis, hai app ufficiale, contro-fa solo on/off , niente tapaprelle dimmer consumi comandi scenari , passa per il cloud e limite di 500 comandi al giorno
4) Sdomotica (pro- hai tutto controllabile, on/off dimmer consumi comandi scenari, comandi ambiente uso dell'app ufficiale e hai la mia assistenza, contro-costa, limite delle 500 chiamate al giorno_

**Reply by 23Pingu23** [reply]
Buongiorno, ho integrato Bticino in HA con integrazione ufficiale Netatmo ed ho notato che le luci dimmer le posso comandare in accensione ed incremento intensità, ma si spegne automaticamente il pulsante in HA mentre la luce non la posso piu ne spegnere ne diminuire intensità se non con controlli Homekit che uso attualmente (o app Control di bticino). Ho letto qui sopra che con l'integrazione nativa dimmer non funziona, ma è di quasi 2 anni questo post e mi chiedevo se fosse ancora valido oppure se ho sbagliato io qualcosa. Le altre luci "normali" funzionano

**Reply by jovannig** [reply]
Scusate, ma ho la living now perfettamente integrata in home assistant, a parte il comando entra esci. Ho provato in tutti i modi da ZHA ma non sono riuscito a farglielo vedere. Sapete se c'e' un modo di farlo oppure sto solo perdendo tempo? Mi trovo con questo interruttore all'ingresso dell'abitazione e non so che farmene. Se proprio non lo posso usare provero' a venderlo e a prendere qualcosa di simile. A proposito, in tal caso quale dispositivo mi consigliate?

**Reply by foice** [reply]
Dal 2021 ad oggi è cambiato qualcosa nei 4 metodi sopra elencati? 

Per contesto: vorrei provare ad usare i comandi connessi per le tapparelle K4027C. Con cosa é compatibile? Io li vedo in Home Assistant come ` Legrand Shutter SW with level control` via Zigbee e funzionano i tre comandi "abbassa", "alza" e "stop". Anche l'interruttore per le luci sembra funzionare. Il dimmer invece non risponde ai comandi, pure se é stato trovato via Zigbee immediatametne.

**Comment by CiccioDroid** [comment]
Grazie infinite per la risposta completa ed esaustiva, purtroppo riporta quanto sono riuscito a trovare online nelle mie ricerche.
Valutando punto per punto:

1. Usando le pennette puoi optare per la soluzione cloud offerta da NabuCasa e comunque avere qualcosa che simula quello che faresti con l'app ufficiale, perdi gli scenari, ma con un pò di lavoro te le crei con HA (pazienza se il Gateway lo devo tenere scollegato).
2. Per me da scartare a priori, a meno che non è un bug segnalato e in corso di risoluzione, in quel caso posso sperare che prima o poi venga risolto
3. Questa è peggio della 2, quindi scartata a priori, sempre tenendo conto che essendo una integrazione magari la potenziano nel tempo
4. A quanto ho capito si parla di qualcosa di esterno ad HA, quindi perderei tutto quello che oggi ho già integrato e inoltre, passare da una soluzione gratis ad una con un costo non mi sembra ottimale visto che c'è comunque NabuCasa.

A questo punto ho altre domande.
Qualcuno ha attivo NabuCasa? nel tempo ho visto che le policy sono cambiate, prima veniva specificato che il loro cloud è sicuro ma gli interni possono vedere il traffico (cosa che mi aveva fatto desistere) adesso viene assicurato che i dati sono protetti (vabbè, ci credo poco, ma già meglio del disclaimer dove mi dicono che se li possono guardare come e quando vogliono). Ovviamente, ho scartato a priori le soluzioni SSL offerte da duckdns e simili, proprio perchè non mi fido molto; motivo per cui uso il cloud di tutti i dispositivi che mi consentono di farlo, senza andare ad integrarli per forza in HA.

Cosa si intende per 500 chiamate al giorno? sono 500 chiamate univoche (quindi una per ogni componente da interrogare) o 500 accessi e quindi ogni volta che apro HA vedo lo stato di prese, relè, tapparelle e consumi con una chiamata?

Uso una pennetta CC2531 (con l'intenzione di fare upgrade verso la CC2652R o la Combee2) e ho letto che usando zigbee2mqtt i device bticino si aggiornano solo all'avvio e non ogni tot secondi e per farlo devo creare un comando di get schedulato. Vi torna questa cosa? Inoltre, con l'uso dell'accoppiata CC2531 e zigbee2mqtt mi capita (almeno una volta al mese) che uno o più sensori perdano la connessione, anche se ho dei repeater abbastanza vicini, l'unico modo per farli riprendere è fare di nuovo il pairing. Questo sarebbe molto scomodo nel caso di device bticino, perchè rifare il pairing ogni volta vorrebbe dire smontare placche o spostare mobili, quindi non proprio comodissimo. Anche su questo avete consigli? Ho già provato a cambiare il Firmware della CC2531 mettendo quello con routing, la situazione è leggermente migliorata, ma si continua a presentare.

Grazie

**Comment by sdomotica** [comment]
Rispondo in quanto Diego mi ha tirato in ballo, anche se la mia soluzione è di fatto scartata. Ma mi permetto di precisare prima qualche cosa e poi di risponderti per quanto di mia conoscenza.
Una "cosa" a pagamento è fatta di prodotto(software) e servizio (know how, assistenza) etc. Si ha una persona a cui chiedere che è eticamente obbligata a rispondere, e non si deve aspettare qualche anima pia su internet che risponda.
Sdomotica è sia un addon per Hassio che un servizio che gira a parte per chi ama HA Core. Ed è un gateway mqtt che permette pertanto di interagire con Living Now attraverso comandi e letture mqtt.
Ma non mi dilungo oltre visto il non interesse.

Passo alle risposte.
L'impostazione delle API ufficiali di Legrand prevede una connessione al loro cloud contingentata. All'inizio erano 100 chiamate, poi grazie alle nostre insistenze sono diventate 500. Io riesco ad arrivare anche a 1.500 ma questo è un altro discorso e lo abilitiamo solo per gli impianti che installiamo noi.

Le 500 sono chiamate sia di lettura stati che di comando. Con Sdomotica abbiamo fatto delle ottimizzazioni. 
Ad esempio se un comando di accensione luce va a buon fine, non facciamo un'altra chiamata per gli stati, ma propaghiamo via mqtt lo stato.
La mia integrazione e quella ufficiale di HA sfrutta questo collegamento, che è quello che Legrand ha previsto per le integrazioni.
Conosco tutte le altre metodologie e tutte hanno pro e contro come sinteticamente esposte da Diego (erano una mia sintesi).

Sono qui se hai altri dubbi, ma solo per l'uso delle API e non per chiavette o Homekit, che non ritengo affidabili per vari motivi.
Sono anche un abbonato a Nabucasa, e ne apprezzo molto la reattività.
Saluti
Sandro

**Reply by CiccioDroid** [reply]
Con me sfondi una porta aperta, ho scelto BTicino (e non sonoff e simili) proprio perchè offre un Harware e il supporto anche se non hai effettivamente acquistato nulla e ti aiutano a capire tutto quello che ti serve per tirare su la tua rete di attuatori e sensori. Questa mia scelta ha alzato non di poco i costi, ma ho roba affidabile, supportata e garantita. Nella mia scelta ho valutato anche sistemi Fibaro e Vimar, ma nessuno di questi era completo e aveva tutti i componenti di cui necessitavo, ma non mi dilungo oltre per non annoiare. 
Insomma, quando parli di prodotto software supportato io ci vado a nozze. Mi scuso per essere stato superficiale, ho aperto velocemente la pagina internet di sdomotica e ho trovato solo informazioni su come integrare BTicino con IPad e avendo un sistema consolidato con componenti di vari produttori ed integrati in HA non volevo cambiare. Ad essere sincero pensavo fosse una soluzione come openHab o smartthings, invece mi sbagliavo.
Ad ogni modo sono una persona che grazie alle anime pie che si trovano su internet ha tirato su roba non da poco e spesso quelle anime ne sanno molto di più di altri che trovandosi a lavorare per questa o quell'azienda, non capiscono nulla di quello che fanno. Questo ovviamente non è il tuo caso a quanto posso notare, ma non sopravvaluterei le cose a pagamento come migliori solo perchè, appunto, si pagano. Bisogna informarsi e capire se il supporto vale la spesa per averlo.
A questo punto, avendo una integrazione con HA e addirittura qualcosa che posso installare con HA Core (suppongo anche una immagine per docker, visto che ho una soluzione HA Core su docker) sono incuriosito dalla soluzione.
Per quanto riguarda chiavette e Homekit nella loro instabilità, sono molto più stabili di tante altre cose, soprattutto quando necessitano di un Gateway che per funzionare deve per forza collegarsi a questo o quel cloud che spesso è lento o irraggiungibile. Infatti uso HA con chiavetta e zigbee2mqtt perchè per adesso la soluzione che funziona solo a casa è più che sufficiente, ma spendere bei soldi per ampliare la rete richiede anche uno studio sugli sviluppi futuri e non voglio precludermi la possibilità di accedere a casa mia anche da remoto.
Per l'uso di API non ho domande, seppur sono uno smanettone e riesco a far tutto quello che mi serve ho preferito affidarmi ad una soluzione da smanettare poco come HA proprio per quel motivo, quindi preferisco continuare ad usare roba confezionata da altri, che sicuro fanno le cose meglio di come le potrei fare io.

**Comment by ganaweb** [comment]
Anche io ho il sistema Living Now della bTicino e mi sono comportato così.
Tramite gateway e l'integrazione Homekit ho collegato ad Home Assistant solamente gli interruttori e deviatori connessi. Questo perché avendo anche un interruttore wireless e un comando Entra&Esci avevo necessità che fossero collegati all'app nativa Home+ Control.
Tutto il resto (prese connesse e Modulo DIN gestione carichi) l'ho collegati tramite Zigbee2MQTT e chiavetta ConBee II. Questo mi ha permesso di gestirne anche i carichi.
Il sistema per ora è stabile e non ho mai avuto disconnessioni.

**Reply by Giampy72** [reply]
Ciao Ganaweb
Vorrei fare la stessa cosa anche io con gli interruttori si home assistant ma quando integro con homekit mi crea solo un bridge e non so come creare le entita’ . La configurazione dice di controllare nelle notifiche dove vedo un qrcode che pero’ se non ho capito male serve per integrare il tutto con l’app Casa di Apple.
Dove sto sbagliando? Salto qualche passaggio?
Premetto che dall’app Casa di Apple ho dissociato il gateway bticino visto che ho letto in altri post che occorre fare questo passaggio.
Ad ogni modo sembra che home assistant crei un bridge ma non vede gli interruttori .
Mentre per le prese connesse e il modulo gestione consumi e carichi allora meglio passare a chiavetta conbee?

**Comment by CiccioDroid** [comment]
Grazie infinite per la risposta, quindi questo conferma che ancora oggi l'integrazione homekit ha il bug sui consumi.
Stavo pensando anche io ad una soluzione del genere, ma non avendo necessità di avere tutto collegato ad internet sto seriamente pensando di collegare tutto direttamente alla chiavetta e tagliare la testa al toro.
A questo proposito approfitto della tua gentilezza per chiederti come ti trovi con la conbee2. La mia CC2531 ormai non ce la fa più e mi serve qualcosa di più affidabile; purtroppo la CC2652R sembra introvabile, quindi mi serve qualcosa di affidabile e che possa seguire l'evoluzione della rete. Hai avuto rogne nel configurare il tutto?

**Reply by ganaweb** [reply]
Scusa per il ritardo con cui ti rispondo. Io con la ConBee II mi trovo benissimo, mai una disconnessione. Ho avuto solo problemi quando ho attivato il TRIM su SSD. Non funzionava più, sapevo che nel mio caso, serviva collegarla tramite prolunga USB ma, nonostante ne avessi acquistata una da due metri, il sistema rilevava continuamente connessioni e disconnessioni nel giro di pochi secondi. Tolto il TRIM e riportato tutto su scheda SD è tornata a funzionare non dandomi mai nessun problema.

Io ho voluto mantenere le luci tramite APP nativa perché avendo insieme al gateway il comando entra e esci, ho mantenuto la possibilità di usarlo per le luci (e per altri dispositivi collegati a questa funzione tramite automazioni). Cosa intendi per tutto collegato in internet? Anche se colleghi il tutto tramite chiavetta zigbee bypassando il gateway, home assistant ha comunque la possibilità di essere esposto in internet ed essere gestito da remoto, quindi in realtà il gateway non serve a nulla nel nostro caso.

**Reply by CiccioDroid** [reply]
Grazie, mi hai convinto, ho appena presa la combee2.
Con tutto connesso ad internet intendo che non ho bisogno di accedere per forza alla casa in remoto, quindi posso anche collegare tutto alla pennetta e in caso abbonarmi a nabucasa (come dicevo in precedenza duckdns e simili non mi convincono).

**Comment by CiccioDroid** [comment]
Ho segnato il thread come risolto.
Tirando le somme, la soluzione migliore è collegare i dispositivi direttamente ad HA con Conbee2+zigbee2mqtt.
Per adesso sono riuscito a fare l'upgrade verso la Conbee2 e devo dire che le cose sono migliorate parecchio, adesso le connessioni sono più stabili e (mi sembra) che i comandi vengono gestiti più velocemente, prima avevo un leggero lag tra pressione e attivazione.

**Comment by gurreri** [comment]
Ciao a tutti
io ho un problema e spero che qualcuno si sia già imbatutto in ciò per una soluzione, vi spiego:
Ho comprato della bticino il gateway e modulo entra/esci wirelles per integrarli in HA creando gli scenari premendo il tasto entra/esci
ma in HA verso il gateway e vedo del comando wirelles lo stato batteria e il button per indentificarlo, ma non vedo da quello che mi aspettavo le due entità Entra e Esci a cui associare una scena.
Qualcuno h risolto?
grazie mille
sandro

**Reply by nolith** [reply]
hai mai risolto @gurreri? Sono nella tua stessa situazione, ma io nemmeno vedo lo stato batteria ed il bottone per l'identificazione.

Vorrei prendere un altro comando luci wireless per fare delle configurazioni ma poi ho paura che sia invisibile in HA anche lui

**Comment by lorekojac** [comment]
Ciao, Probabilmente farò una domanda stupida, ma sono piuttosto nuovo nel mondo di home assistant.
Ho la serie Living now e con l'app ufficiale ho collegato tutto ma noto dei ritardi importanti nella risposta di home assistant a un'accensione fisica da pulsante di una luce. Vorrei migrare tutto su Homekit device ma non riesco a collegarlo (C'ero riuscito in modo casuale tempo fa), E' possibile farlo senza un dispositivo apple? C'è una procedura particolare?
Grazie

**Reply by robyseve78** [reply]
Anche io ho il problema del ritardo all'accensione fisica. Tu hai risolto?

**Reply by maxchianese** [reply]
Anche io ho lo stesso problema …. Avete risolto???

**Reply by maxchianese** [reply]
Accensione dal tasto fisico, ritardo su home Assistant

**Reply by Mec0808** [reply]
La latenza è un problema dell'integrazione BTicino, avevo inserito BTicino in home assistant anche con l'integrazione homekit, ma ora non funziona più, peccato perché con homekit lavorava in locale senza l'ausilio del Cloud

**Reply by lorekojac** [reply]
Purtroppo ancora nessuna novità..

**Reply by sdomotica** [reply]
Se avete problemi con Homekit e con l'integrazione nativa (ambedue gratuite) io ho realizzato un addon per Raspberry (trovate citato il mio lavoro sopra, che però è a pagamento) che usando le stesse API dell'integrazione nativa, aggiorna più spesso (10 secondi) ed ha più funzionalità
![Immagine WhatsApp 2024-09-30 ore 17 19 31_41fb3ed9](https://github.com/user-attachments/assets/100247b7-0220-47d3-ba79-e1c08de98eda)

![Immagine WhatsApp 2024-09-30 ore 17 19 31_d29745d2](https://github.com/user-attachments/assets/e00bba3c-3264-4a01-878b-6962637b62d8)

**Comment by maxchianese** [comment]
Ciao Quanto costa sistemare il tutto?Inviato da iPhone15proMax Il giorno 30 set 2024, alle ore 17:21, Sandro Di Marco ***@***.***> ha scritto:﻿
Se avete problemi con Homekit e con l'integrazione nativa (ambedue gratuite) io ho realizzato un addon per Raspberry (trovate citato il mio lavoro sopra, che però è a pagamento) che usando le stesse API dell'integrazione nativa, aggiorna più spesso (10 secondi) ed ha più funzionalità
Immagine.WhatsApp.2024-09-30.ore.17.19.31_41fb3ed9.jpg (view on web)
Immagine.WhatsApp.2024-09-30.ore.17.19.31_d29745d2.jpg (view on web)

**Reply by sdomotica** [reply]
Puoi scrivermi direttamente sandro.dimarco chiocciola   NO LINK oppure mi trovi su Facebook dove puoi vedere tutti i progetti.
NO LINK
Ciao
Sandro
